src/ORM/Connect/DBSchemaManager.php

```diff
@@ -887,6 +894,17 @@ public function alterationMessage($message, $type = "")
         }
     }
 
+    /**
+     * Take the column spec and convert it into the spec for a generated column.
+     */
+    public function makeGenerated(string $spec, array $origSpec, string $expression, string $generationType): string
+    {
+        // Just return the original spec, for BC.
+        // No value will actually get generated if subclasses don't implement this method, so this will just
+        // be taking space in the DB for no value but at least the site won't just fail to build altogether.
+        return $spec;
```

Comment:
If we didn't do this here, the only alternatives I could think of were:
1. Check `method_exists()` and if it doesn't, log a warning and don't create the field. But then code that expects the field (either explicitly or implicitly) would either break or need updating to handle the case where it isn't there.
2. Check `method_exists()` and throw an exception if it doesn't. But that's effectively the same as implementing this as `abstract` which is a API breaking change.

Reply:
The same applies for other new methods in this class.